src/contract/table-cursor.ts

```diff
@@ -63,113 +55,65 @@ export class TableCursor<TableRow> {
                 yield row
             }
 
-            // If no rows are returned or next_key is undefined, we have exhausted all rows
             if (rows.length === 0 || !this.next_key) {
                 return
             }
         }
     }
 
-    /**
-     * Fetches more rows from the table and appends them to the cursor.
-     *
-     * @returns The new rows.
-     */
-    async next(rowsPerAPIRequest?: number): Promise<TableRow[]> {
+    async next(rowsPerAPIRequest?: number): Promise<RowType[]> {
         if (this.endReached) {
             return []
         }
 
-        let lower_bound = this.tableParams.lower_bound
-        const upper_bound = this.tableParams.upper_bound
+        let lower_bound = this.params.lower_bound
+        const upper_bound = this.params.upper_bound
 
         if (this.next_key) {
             lower_bound = this.next_key
         }
 
-        let indexPosition = this.tableParams.index_position || 'primary'
-
-        if (this.indexPositionField) {
-            const fieldToIndexMapping = this.table.getFieldToIndex()
-
-            if (!fieldToIndexMapping[this.indexPositionField]) {
-                throw new Error(`Field ${this.indexPositionField} is not a valid index.`)
-            }
-
-            indexPosition = fieldToIndexMapping[this.indexPositionField].index_position
-        }
-
-        const result = await this.table.contract.client!.v1.chain.get_table_rows({
-            ...this.tableParams,
-            limit: Math.min(
-                this.maxRows - this.rowsCount,
-                rowsPerAPIRequest || this.tableParams.limit
-            ),
+        const query = {
+            ...this.params,
+            limit: Math.min(this.maxRows - this.rowsCount, rowsPerAPIRequest || this.params.limit),
             lower_bound: wrapIndexValue(lower_bound),
             upper_bound: wrapIndexValue(upper_bound),
-            index_position: indexPosition,
-            json: false,
-        })
+        }
 
-        let {rows} = result
-        this.next_key = result.next_key
+        const result = await this.client!.v1.chain.get_table_rows(query)
 
+        const rows =
```

Comment:
This statement variable assignment does a lot and is a bit difficult to follow so I would probably break it up into different steps.

Reply:
I gave this a shot, and added some comments. 

https://github.com/wharfkit/contract/blob/ede31bfd05b00139969f82b06968d98b204b2f12/src/contract/table-cursor.ts#L85-L100

I'm not sure how much more I think we should split it out, unless we wanted to get rid of the ternary operator, and instead do... 

```ts
let rows = thing 
if (decode) { 
  rows = otherThing 
}
```